src/test/java/org/inferred/freebuilder/processor/NullableSourceTest.java

```diff
@@ -0,0 +1,490 @@
+        "    setName(template.getName());",
+        "    setAge(template.getAge());",
+        "    return (Person.Builder) this;",
+        "  }",
+        "",
+        "  /**",
+        "   * Resets the state of this builder.",
+        "   */",
+        "  public Person.Builder clear() {",
+        "    Person_Builder _template = new Person.Builder();",
+        "    name = _template.name;",
+        "    age = _template.age;",
+        "    return (Person.Builder) this;",
+        "  }",
+        "",
+        "  /**",
+        "   * Returns a newly-created {@link Person} based on the contents of the {@code Builder}.",
+        "   */",
+        "  public Person build() {",
+        "    return new Person_Builder.Value(this);",
+        "  }",
+        "",
+        "  /**",
+        "   * Returns a newly-created partial {@link Person}",
+        "   * based on the contents of the {@code Builder}.",
+        "   * State checking will not be performed.",
+        "   *",
+        "   * <p>Partials should only ever be used in tests.",
+        "   */",
+        "  @VisibleForTesting()",
+        "  public Person buildPartial() {",
+        "    return new Person_Builder.Partial(this);",
+        "  }",
+        "",
+        "  private static final class Value extends Person {",
+        "    @Nullable private final String name;",
+        "    @Nullable private final Integer age;",
+        "",
+        "    private Value(Person_Builder builder) {",
+        "      this.name = builder.name;",
+        "      this.age = builder.age;",
+        "    }",
+        "",
+        "    @Override",
+        "    @Nullable",
+        "    public String getName() {",
+        "      return name;",
+        "    }",
+        "",
+        "    @Override",
+        "    @Nullable",
+        "    public Integer getAge() {",
+        "      return age;",
+        "    }",
+        "",
+        "    @Override",
+        "    public boolean equals(Object obj) {",
+        "      if (!(obj instanceof Person_Builder.Value)) {",
+        "        return false;",
+        "      }",
+        "      Person_Builder.Value other = (Person_Builder.Value) obj;",
+        "      if (name != other.name && (name == null || !name.equals(other.name))) {",
+        "        return false;",
+        "      }",
+        "      if (age != other.age && (age == null || !age.equals(other.age))) {",
+        "        return false;",
+        "      }",
+        "      return true;",
+        "    }",
+        "",
+        "    @Override",
+        "    public int hashCode() {",
+        "      return Arrays.hashCode(new Object[] {name, age});",
+        "    }",
+        "",
+        "    @Override",
+        "    public String toString() {",
+        "      return \"Person{\"",
+        "          + COMMA_JOINER.join(",
+        "              (name != null ? \"name=\" + name : null), " +
+            "(age != null ? \"age=\" + age : null))",
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [The String literal "(age != null ? \"age=\" + age : null))" appears 4 times in this file; the first occurrence is on line 187](https://www.codacy.com/app/Chris-Purcell-39/FreeBuilder/file/2277828626/issues/source?bid=3146243&fileBranchId=3187161#l187)